Add wmp checker and remove the duplicated source in spec file

nick-wang opened the pull request:
This pull request including two enhancements:
1) Fix https://github.com/SUSE/sapwmp/issues/4
  Move the `wmp_check` script from [independent repo](https://github.com/scmschmidt/wmp_check/) to here to ease the checker management. Tested against SLE15SP0/SP1/SP2 already.
2) Remove the duplicated source in .spec file, install directly from the source code. Tested on SLE15SP2, installed rpm including files:
```
/etc/sapwmp.conf
/usr/lib/sapwmp
/usr/lib/sapwmp/sapwmp-capture
/usr/lib/sapwmp/scripts
/usr/lib/sapwmp/scripts/README-wmp-check.md
/usr/lib/sapwmp/scripts/wmp-check.sh
/usr/lib/sapwmp/wmp-sample-memory
/usr/lib/supportconfig
/usr/lib/supportconfig/plugins
/usr/lib/supportconfig/plugins/sapwmp
/usr/lib/systemd/system/SAP.slice
/usr/lib/systemd/system/wmp-sample-memory.service
/usr/lib/systemd/system/wmp-sample-memory.timer
```

nick-wang committed:
Include a script to check the setup of wmp.

Fix issue request https://github.com/SUSE/sapwmp/issues/4

Include the wmp check script for a preflight check of wmp setup.
Merge from https://github.com/scmschmidt/wmp_check/, version 1.1.1

nick-wang committed:
Modify the spec file to include script and remove duplicated files.

scmschmidt reviewed (commented):
Looks good to me.

Werkov commented on rpm/sapwmp.spec:
```diff
@@ -71,12 +65,17 @@ Configuration and utilities for collecting SAP processes under control group to
-install -D -m 744 %{SOURCE4} %{buildroot}/%{_libexecdir}/sapwmp/wmp-sample-memory
-install -D -m 644 %{SOURCE5} %{buildroot}/%{_unitdir}/wmp-sample-memory.service
-install -D -m 644 %{SOURCE6} %{buildroot}/%{_unitdir}/wmp-sample-memory.timer
+%define wmpd %{_builddir}/%{name}-%{version}
```
Interesting trick to get rid of %{SOURCEn}. IIUC, this also implies the `extract_file` service can be dropped from OBS `_service` file. (Not relevant to this repo, but just mentioning it.)

nick-wang replied:
Yes, `extract_file` no longer need in `_service` file any more. (Without you mentioning this, I didn't notice that part existed)

Will align the removed source and _service files in ibs after this pull request merged.

Werkov commented on rpm/sapwmp.spec:
```diff
@@ -71,12 +65,17 @@ Configuration and utilities for collecting SAP processes under control group to
+
+mkdir -p %{buildroot}/%{_libexecdir}/sapwmp/scripts
+install -D -m 755 %{wmpd}/scripts/*.sh %{buildroot}/%{_libexecdir}/sapwmp/scripts/
+install -D -m 644 %{wmpd}/scripts/README* %{buildroot}/%{_libexecdir}/sapwmp/scripts/
```
It's a judgement call whether to distribute the README (note we also have [documentation](https://documentation.suse.com/sles-sap/15-SP2/html/SLES-SAP-guide/cha-tune.html#sec-memory-protection)). However, if doing so, it seems `/usr/share/doc/packages` is more standard location.

Werkov commented on rpm/sapwmp.spec:
```diff
@@ -71,12 +65,17 @@ Configuration and utilities for collecting SAP processes under control group to
+install -D -m 755 %{wmpd}/rpm/supportconfig-sapwmp %{buildroot}%{_prefix}/lib/supportconfig/plugins/sapwmp
+install -D -m 744 %{wmpd}/rpm/wmp-sample-memory.sh %{buildroot}/%{_libexecdir}/sapwmp/wmp-sample-memory
+install -D -m 744 %{wmpd}/rpm/wmp-sample-memory.service %{buildroot}/%{_unitdir}/wmp-sample-memory.service
+install -D -m 744 %{wmpd}/rpm/wmp-sample-memory.timer %{buildroot}/%{_unitdir}/wmp-sample-memory.timer
```
The service/timer units should be 644, not executable.

Werkov commented on rpm/sapwmp.spec:
```diff
@@ -127,5 +126,6 @@ fi
 %dir %{_prefix}/lib/supportconfig
 %dir %{_prefix}/lib/supportconfig/plugins
 %{_prefix}/lib/supportconfig/plugins/sapwmp
+%{_libexecdir}/sapwmp/scripts
```
Missing `%dir`?

nick-wang replied:
I remove `%dir` on purpose, since this folder will be used for `sapwmp` package only. Without `%dir`, it will package all the files belong to this folder as well.

Werkov replied:
Of course, you're right, my RPM-fu was weak here. I'll mark this as resolved.

Werkov commented on scripts/README-wmp-check.md:
````diff
@@ -0,0 +1,165 @@
+
+## Usage
+```
+wmp-check.sh
````
This usage assume the script is in `$PATH`. Perhaps I wouldn't put the script to `$PATH` but instead referenced to it with full path.

Werkov reviewed (changes requested):
Distributing the script seems useful in general. I've added just some minor nitpicks inline.

nick-wang committed:
spec: polish spec file based on review comments.

Move README to /usr/share/doc/packages/sapwmp
Remove the executable bit of service/timer file
Use absolute path for wmp-check.sh

nick-wang commented:
Rebased the pull request based on the new merged codes.("capture: Graceful handling of missing PARENT_COMMANDS" and "Configurable properties")

scmschmidt commented:
Just noticed: /usr/lib/sapwmp/scripts/wmp-check.sh

Why do not put it in /usr/sbin/, so it is in the PATH variable and can be reached easily?

nick-wang commented:
> Just noticed: /usr/lib/sapwmp/scripts/wmp-check.sh
> 
> Why do not put it in /usr/sbin/, so it is in the PATH variable and can be reached easily?

@scmschmidt  A similar conversation about the script PATH just happened in this pull request, please refer to review comment above.

I was thinking to put it together with `/usr/lib/sapwmp/sapwmp-capture`. But it also make sense to put in $PATH without the ".sh", since it is a standalone script instead of library.

@Werkov How do you think of putting `wmp-check` to `/sbin`? (`/usr/sbin` may not suitable, due to the permission of `/boot/grub2/grub.cfg` and usage of `/usr/sap/*/SYS/profile/*`require need root privileges.)

scmschmidt commented:
I was almost sure, I had read a discussion about it. Sorry.

Werkov commented:
I'd decide based on the use case. If the check is supposed to called from various scripts (such as supportconfig or some QA automation), I wouldn't expose it in PATH. If there's a use case to run it from a user session (I'd listen to @scmschmidt here), then there's no benefit in hiding.

I don't know what's the difference between `/sbin` and `/usr/sbin`. If you meant `/usb/bin` vs `/usr/sbin`, I agree the latter sounds better given the permissions.

scmschmidt commented:
I have the tool mentioned in the documentation as a way for the customer to check the WMP setup.

nick-wang commented:
> I have the tool mentioned in the documentation as a way for the customer to check the WMP setup.

Got it. Based on the scenario and the discussion above, i will put `wmp-check.sh` into `/sbin` with privileges. Likly remove the `.sh` as most script did in `/usr/sbin`

nick-wang committed:
Install the wmp-check to /usr/sbin

Werkov reviewed (commented):
Almost perfect but sorry to be a nuisance once more -- is there a reason to switch from dash to underscore in the tool name? (There are already dashes in sapwmp-capture itself or the wmp-sample-memory.)

nick-wang commented:
> Almost perfect but sorry to be a nuisance once more -- is there a reason to switch from dash to underscore in the tool name? (There are already dashes in sapwmp-capture itself or the wmp-sample-memory.)

@Werkov Actually both of them are OK.

The original script name is with underscore like `wmp_check`. I modified to `wmp-check.sh` due to other script like `wmp-sample-memory.sh` are using dash. To unified them, i replaced to `wmp-check.sh`.

However, after move the script to `/usr/sbin`. I found the folder mixed with scripts with dash and underscore... And no reason to align with `wmp-sample-memory.sh` anymore since they are not in the same place. So i change back to `wmp_check` again to avoid modification in other place. (@scmschmidt mentioned the tool in documentations as well)

Werkov commented:
The [documentation](https://documentation.suse.com/sles-sap/15-SP2/html/SLES-SAP-guide/cha-tune.html#sec-memory-protection) refers to a [separate repo](https://github.com/scmschmidt/wmp_check), I assume it should be updated once the check script is moved into the package (assuming it's going to be "maintained" here), therefore dashes seem more consistent to me (but I'm not losing my sleep over it ;-)

nick-wang commented:
@Werkov I revert the name to `wmp-check` and force based the last commit.

scmschmidt commented:
> The [documentation](https://documentation.suse.com/sles-sap/15-SP2/html/SLES-SAP-guide/cha-tune.html#sec-memory-protection) refers to a [separate repo](https://github.com/scmschmidt/wmp_check), I assume it should be updated once the check script is moved into the package (assuming it's going to be "maintained" here), therefore dashes seem more consistent to me (but I'm not losing my sleep over it ;-)

Hi Michal. Documentation change is already prepared. As soon as the package is out, the documentation gets updated accordingly to reflect all the changes we made. ;-)

nick-wang commented:
As long as this pull request and https://github.com/SUSE/sapwmp/pull/12 (need rebase)  merged, i will submit new change this into sp3/sp2/sp1.

Werkov reviewed (approved):
LGTM. Thanks!

nick-wang commented:
@Werkov @scmschmidt Thanks for the review and the valuable comments!!
